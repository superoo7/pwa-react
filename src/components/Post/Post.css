.title {
  font-size: 3rem;
  color: #2F89DE;
}

.content {
  font-size: 1.8rem;
  line-height: 1.8;
}

.link {
  color: #2F89DE;
}

.link:focus,
.link:hover {
  color: #30699E;
  transition: color 0.3s ease;
}
