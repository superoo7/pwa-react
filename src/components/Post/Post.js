import React from 'react'
import styles from './Post.css'

import { Link } from 'react-router-dom'
const Post = (props) => (
  <div>
    <h2 className={styles.title}>
      { props.titleLink
      ? <Link className={styles.link} to={`/post/${props.slug}`}>{props.title}</Link>
      : props.title
      }
    </h2>
    <p className={styles.content}>
      {props.excerpt}
    </p>
  </div>
)

export default Post
