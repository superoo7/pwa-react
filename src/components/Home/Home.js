import React from 'react'
import Post from '../Post/Post'

// console.log(posts)

const Home = ({posts}) => (
  <div>
    {
      posts.length ?
      posts.map(post => <Post {...post} key={post.slug} titleLink />) :
      <p>Awaiting post</p>
    }
  </div>
)

export default Home
