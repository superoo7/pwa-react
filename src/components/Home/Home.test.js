import React from 'react'
import { shallow } from 'enzyme'
import Home from './Home'
import Post from '../Post/Post'
import toJson from 'enzyme-to-json'


test('Home component should render as expected', () => {
  const posts = [{
    "title": "My experience with EE 1",
    "slug": "my-experience-with-ee-1",
    "excerpt": "Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed",
    "content": "do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolor"
  }]
  const component = shallow(<Home posts={posts} />)
  const tree = toJson(component)
  // console.log(tree)
  expect(component.find(Post).length).toBe(1)
  expect(tree).toMatchSnapshot()
})


test('Home component should render as expected', () => {
  const posts = [{
    "title": "My experience with EE 1",
    "slug": "my-experience-with-ee-1",
    "excerpt": "Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed",
    "content": "do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolor"
  },
  {
    "title": "My experience with EE 1",
    "slug": "my-experience-with-ee-1",
    "excerpt": "Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed",
    "content": "do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolor"
  }]
  const component = shallow(<Home posts={posts} />)
  const tree = toJson(component)
  // console.log(tree)
  expect(component.find(Post).length).toBe(2)
  expect(tree).toMatchSnapshot()
})

test('Home component should render as expected', () => {
  const posts = []
  const component = shallow(<Home posts={posts} />)
  const tree = toJson(component)
  // console.log(tree)
  expect(component.find(Post).length).toBe(0)
  expect(tree).toMatchSnapshot()
})
