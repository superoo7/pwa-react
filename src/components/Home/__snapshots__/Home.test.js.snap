// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home component should render as expected 1`] = `
<div>
  <Post
    content="do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolor"
    excerpt="Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed"
    key="my-experience-with-ee-1"
    slug="my-experience-with-ee-1"
    title="My experience with EE 1"
    titleLink={true}
  />
</div>
`;

exports[`Home component should render as expected 2`] = `
<div>
  <Post
    content="do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolor"
    excerpt="Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed"
    key="my-experience-with-ee-1"
    slug="my-experience-with-ee-1"
    title="My experience with EE 1"
    titleLink={true}
  />
  <Post
    content="do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolor"
    excerpt="Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed"
    key="my-experience-with-ee-1"
    slug="my-experience-with-ee-1"
    title="My experience with EE 1"
    titleLink={true}
  />
</div>
`;

exports[`Home component should render as expected 3`] = `
<div>
  <p>
    Awaiting post
  </p>
</div>
`;
